src/mmg3d/hash_3d.c

```diff
@@ -1837,10 +1825,71 @@ int MMG5_chkBdryTria(MMG5_pMesh mesh) {
     fprintf(stderr,"\n  ## Warning: %s: %" MMG5_PRId " extra boundaries founded\n",
             __func__,nbl);
 
-  /* Fill missing bdy triangles */
-  return MMG5_bdryTria(mesh,ntmesh);
 }
 
+int MMG5_chkbdryTria_deleteExtraBoundaries() {
+  nt=0; nbl=1;
+
+  if ( ! MMG5_hashNew(mesh,&hashTri,0.51*mesh->nt,1.51*mesh->nt) ) return 0;
+
+  for (k=1; k<=mesh->nt; k++) {
+    ptt = &mesh->tria[k];
+
+    ia = ptt->v[0];
+    ib = ptt->v[1];
+    ic = ptt->v[2];
+
+    i = MMG5_hashGetFace(&hashElt,ia,ib,ic);
+    j = MMG5_hashFace(mesh,&hashTri,ia,ib,ic,k);
+
+    ptt->cc = i;
+
+    if ( !j ) {
+      MMG5_DEL_MEM(mesh,hashElt.item);
+      MMG5_DEL_MEM(mesh,hashTri.item);
+      return 0;
+    }
+    else if ( j > 0 ) {
+      /* the face already exists in the tria table */
+      continue;
+    }
+
+    if ( !i ) {
+      /* the triangle is not a boundary tri or a tri at the interface of two
+        * subdomains with different references and the user don't ask to keep
+        * it. */
+      continue;
+    }
+
+    if ( mesh->info.opnbdy ) {
+      kk    = i/4;
+      iface = i%4;
+      adj   = mesh->adja[4*(kk-1)+1+iface];
+      /* Check if we have found a triangle at the interface of 2 doms of same
+        * ref */
+      if ( adj && mesh->tetra[kk].ref == mesh->tetra[adj/4].ref ) {
+        ++ntpres;
+      }
+    }
+
+    ++nt;
+    if ( k!=nbl ) {
+      pttnew = &mesh->tria[nbl];
+      memcpy(pttnew,ptt,sizeof(MMG5_Tria));
+    }
```

Comment:
I think that we can remove / extract this from the loop and just delete extra triangles inside the loop.
Then, after the loop (so outside the `MMG5_chkbdryTria_deleteExtraBoundaries()` func), it should be possible to : 
  - travel triangles
  - detect the deleted tria
  - pack the array accordingly
  - update the perm array for ParMmg only

Reply:
something like that : 

```
nt=0; nbl=1;
for (k=1; k<=mesh->nt; k++) {
    ptt = &mesh->tria[k];
    
    if ( !MG_EOK(ptt) ) continue;
    
    ++nt;
    if ( k!=nbl ) {
      pttnew = &mesh->tria[nbl];
      memcpy(pttnew,ptt,sizeof(MMG5_Tria));
      if ( permtria ) {
         permtria[k] = nbl;
       }
    }
}

Reply:
Or more elegant but also more error-prone (I am not sure that it works ;-) )

```
    k = 1;
    do {
      ptt = &mesh->tria[k];
      if ( !MG_EOK(ptt) ) {
        pttnew = &mesh->tria[mesh->nt];
        assert( ptt && pttnew && MG_EOK(pttnew) );
        memcpy(ptt,pttnew,sizeof(MMG5_Tria));

       if ( permtria ) permtria[mesh->nt] = k;

        ptt->v[0] = 0;
        mesh->nt--;
      }
    }
    while ( ++k < mesh->nt );

```